internal/dataplane/kong_client.go

```diff
@@ -460,7 +460,23 @@ func (c *KongClient) sendOutToClients(
 	ctx context.Context, s *kongstate.KongState, formatVersion string, config sendconfig.Config,
 ) ([]string, error) {
 	clients := c.clientsProvider.AllClients()
+	if len(clients) == 0 {
+		c.logger.Debugf("no available kong clients, skipping sending out configurations")
+		return c.SHAs, nil
+	}
 	c.logger.Debugf("sending configuration to %d clients", len(clients))
+	// send to only one client if gateway is DB-backed.
```

Comment:
We have to take into account that `AllClients` returns the Konnect client if configured. I'd propose changing this to use `GatewayClients()` and adding a `KonnectClient() (*adminapi.Client, bool)` method to handle Konnect separately always.

Before this, the separation wasn't really needed and we could treat all clients (including Konnect) uniformly, but now it changes.

Reply:
I'm pretty sure this was brought up somewhere: do we have to send it to the elected leader or we just pick one, arbitrary instance that will receive the config and we're good?

Can we expect any issues if we send config at point in time N, to instance X, and in N+1 we send it to instance Y?

Reply:
According to the [Admin API docs](https://docs.konghq.com/gateway/latest/admin-api/) that @randmonkey has pointed me to, we can call an arbitrary instance:

> Requests to the Admin API can be sent to any node in the cluster, and Kong will keep the configuration consistent across all nodes.

Reply:
Moved to method `sendOutToClientsDBMode` and in the method we send config to ALL konnect nodes(could be 0 or 1 clients in current state), and ONE gateway client.